languagetool: 5.1 -> 5.2

TredwellGit opened the pull request:
###### Motivation for this change
https://github.com/languagetool-org/languagetool/blob/v5.2/languagetool-standalone/CHANGES.md

###### Things done
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/tools/text/languagetool/default.nix:
```diff
@@ -2,34 +2,32 @@
   installPhase = ''
-    mkdir -p $out/share
-    mv * $out/share/
+    mkdir -p "$out/share"
```
```suggestion
    mkdir -p $out/share
```

SuperSandro2000 commented on pkgs/tools/text/languagetool/default.nix:
```diff
@@ -2,34 +2,32 @@
-    mkdir -p $out/share
-    mv * $out/share/
+    mkdir -p "$out/share"
+    mv -- * "$out/share/"
```
```suggestion
    mv . $out/share/
```

SuperSandro2000 commented on pkgs/tools/text/languagetool/default.nix:
```diff
@@ -2,34 +2,32 @@
-    for lt in languagetool{,-commandline,-server};do
-      makeWrapper ${jre}/bin/java $out/bin/$lt \
+    for lt in languagetool{,-commandline,-server}; do
+      makeWrapper "${jre}/bin/java" "$out/bin/$lt" \
```
This is not required.

SuperSandro2000 commented on pkgs/tools/text/languagetool/default.nix:
```diff
@@ -2,34 +2,32 @@
     done
 
-    makeWrapper ${jre}/bin/java $out/bin/languagetool-http-server \
+    makeWrapper "${jre}/bin/java" "$out/bin/languagetool-http-server" \
```
This neither.

TredwellGit commented:
Not required but it is better Bash.
https://github.com/koalaman/shellcheck/wiki/SC2086
https://github.com/koalaman/shellcheck/wiki/SC2035

TredwellGit commented:
https://github.com/NixOS/nixpkgs/pull/105641

SuperSandro2000 commented:
> #105641

That is for sh files which are often writte in sh. If we would add " to packages we would need to change all packages specifying any phase, pre or post.

TredwellGit committed:
languagetool: 5.1 -> 5.2

https://github.com/languagetool-org/languagetool/blob/v5.2/languagetool-standalone/CHANGES.md

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108912` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>languagetool</li>
<li>vimPlugins.lens-vim</li>
<li>vimPlugins.vim-grammarous</li>
</ul>
</details>
